### ISSUE TITLE
How can exploit parallelism in Optuna? 

### ISSUE THREAD

**Issue by ali2iptoki** [optuna/optuna#2340]
`SuccessiveHalvingPruner` is supposed to be executed in parallel way. How can configure our code (simple code) to exploit the exploit parallelism in Optuna? 

## Note to the questioner

If you are more comfortable with [Stack Overflow], you may consider posting your question with the "optuna" tag there instead.
Alternatively, for issues that would benefit from more of an interactive session with the developers,
you may refer to the [optuna/optuna chat] on Gitter.

[Stack Overflow]: https://stackoverflow.com/questions/tagged/optuna
[optuna/optuna chat]: https://gitter.im/optuna/optuna

**Comment by himkt** [comment]
You can run an optimization in parallel by simply executing your Python script from multiple terminals.
It needs the shared RDB (MySQL or PostgreSQL are recommended). Please see the [documentation](https://optuna.readthedocs.io/en/stable/tutorial/10_key_features/004_distributed.html) for more detail.

**Comment by ali2iptoki** [comment]
@himkt  thanks for your answer. But in the case of executing the same script with the same study-name and storage both times, and if we already configured Optuna with Neptune, we will see two project ID in Neptune, NO?

**Comment by himkt** [comment]
@ali2iptoki 

1.
You can skip creating a new study if it already exists. Please specify `load_if_exists=True` in `create_study`.
(ref. https://optuna.readthedocs.io/en/stable/reference/generated/optuna.study.create_study.html#optuna.study.create_study)

2.
You're right. You have to run a program on each computer.

You can also specify `n_jobs` in `study.optimize`. It would be helpful if you have a machine with a large computing capacity. (ref. https://optuna.readthedocs.io/en/stable/reference/generated/optuna.study.Study.html#optuna.study.Study.optimize)

**Comment by ali2iptoki** [comment]
@himkt  thanks but when I get the following
```
    study = optuna.create_study(
  File "C:\Users\ali\miniconda3\envs\iptoki\lib\site-packages\optuna\study.py", line 1055, in create_study
    storage = storages.get_storage(storage)
  File "C:\Users\ali\miniconda3\envs\iptoki\lib\site-packages\optuna\storages\__init__.py", line 27, in get_storage   
    return _CachedStorage(RDBStorage(storage))
  File "C:\Users\ali\miniconda3\envs\iptoki\lib\site-packages\optuna\storages\_rdb\storage.py", line 163, in __init__
    raise ImportError(
ImportError: Failed to import DB access module for the specified storage URL. Please install appropriate one.
```

I am using mysql

**Comment by himkt** [comment]
The error shows you failed to connect MySQL server.
You have to install and launch a server first.
Then make sure that you can connect a server by MySQL CLI.
Let me put the [reference](https://dev.mysql.com/doc/refman/8.0/en/installing.html) to MySQL documentation.

After you succeed to set up a MySQL server, create a database.
(ref. https://optuna.readthedocs.io/en/latest/tutorial/10_key_features/004_distributed.html)

You can also use other backends such as SQLite, PostgreSQL, and Redis.
I recommend you to use MySQL or PostgreSQL for distributed optimization.

**Comment by Stephocke** [comment]
I have another question regarding parallel computation.
I also had some trouble setting up my MySQL stuff but eventually, I made it. 

Even while I was busy setting it up, it occurred to me that if the `suggest` functions are deterministic, it is not true parallelization as you end up with multiple processes evaluating the same point.

Now that I have started my experiment, I have to conclude that this is really the case. Can this be avoided?
